compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1143,9 +1143,11 @@ impl<'a> Parser<'a> {
                     Ok(kind) => kind,
                     Err(kind) => match kind {
                         ItemKind::Const(box ConstItem { ty, expr, .. }) => {
+                            let const_span = Some(span.with_hi(ident.span.lo()))
+                                .filter(|span| span.can_be_used_for_suggestions());
```

Comment:
I suspect this involves more than just a style refactor. Previously, it checked the result of `﻿span.with_lo(xxx)`. Now, it only checks ﻿`span`.

Reply:
prev:

```rs
let next = span.with_hi(ident.span.lo();
if next.can_be_used_for_suggestions() {
    Some(next)
} else {
    None
}
```

now:

```rs
if span.can_be_used_for_suggestions() {
    Some(span.with_hi(ident.span.lo())
} else {
    None
}
```

I anticipate this may result in a slight difference in the output.

Reply:
Ah that, I don't think so, `with_hi` is a pure function.

Reply:
Oh no, I see it now, lol

Reply:
gonna push away my commit and approve it